cmd/kubelet/app/server.go

```diff
@@ -638,7 +646,12 @@ func RunKubelet(kcfg *KubeletConfig, builder KubeletBuilder) error {
 	eventBroadcaster.StartLogging(glog.V(3).Infof)
 	if kcfg.KubeClient != nil {
 		glog.V(4).Infof("Sending events to api server.")
-		eventBroadcaster.StartRecordingToSink(kcfg.KubeClient.Events(""))
+		if kcfg.EventRecordQPS == 0.0 {
+			eventBroadcaster.StartRecordingToSink(kcfg.KubeClient.Events(""), nil)
+		} else {
+			eventBroadcaster.StartRecordingToSink(kcfg.KubeClient.Events(""), util.NewTokenBucketRateLimiter(kcfg.EventRecordQPS, kcfg.EventBurst))
```

Comment:
Terribly sorry that I've been sitting on this, I guess I didn't see that you pushed changes. I have been thinking about this for a bit, and it occurs to me that there's another way to do this.

The solution you have here will work, but there's two downsides:
* People pass `nil` everywhere now.
* It's unpredictable how much quota will be left for non-event client usage.

An idea that fixes both of these is:
1. Make a copy of kcfg.KubeClient
2. Modify the copy's internal RESTClient.Throttle object (I think enough is public that you can do this)
3. Start recording to this copied client.

Then you don't have to add throttling to the event recorder at all, and the event QPS won't count against the regular QPS.

I'm sorry for not thinking of this to begin with. What do you think?